linux/check_inbox_driver/get_inbox_drivers.yml

```diff
@@ -163,42 +189,50 @@
           with_items: "{{ stat_drivers_filenames.results }}"
           loop_control:
             loop_var: stat_driver_filename
-      when: guest_os_ansible_distribution != "VMware Photon OS"
-  when: >
-    not (guest_os_ansible_distribution == "VMware Photon OS" and
-    guest_os_ansible_distribution_major_ver | int < 4)
 
-# If the driver is not found in guest OS, set its version to 'N/A'
-- name: "Update fact of inbox driver version dict to set not found driver version with 'N/A'"
+- name: "Display versions of inbox drivers on {{ vm_guest_os_distribution }}"
+  ansible.builtin.debug: var=inbox_drivers_versions
+
+- name: "Set fact of missing mandatory inbox drivers on {{ vm_guest_os_distribution }}"
   ansible.builtin.set_fact:
-    inbox_drivers_versions: >-
+    missing_mandatory_drivers: >-
       {{
         inbox_drivers_versions |
-        combine({module_name: 'N/A'})
+        dict2items |
+        selectattr('key', 'in', mandatory_drivers) |
+        selectattr('value', 'equalto', 'N/A') |
+        map(attribute='key')
       }}
-  loop: "{{ inbox_drivers | difference(inbox_drivers_versions.keys()) }}"
-  loop_control:
-    loop_var: module_name
 
-- name: "Check all mandatory inbox drivers exists in {{ guest_os_ansible_distribution }}"
-  ansible.builtin.assert:
-    that:
-      - inbox_drivers_versions[item]
-      - inbox_drivers_versions[item] != 'N/A'
-    fail_msg: "Inbox driver {{ item }} is missing in {{ guest_os_ansible_distribution }} {{ guest_os_ansible_distribution_ver }}"
-  with_items: "{{ mandatory_drivers }}"
-  when: >
-    not (guest_os_ansible_distribution == "VMware Photon OS" and
-    guest_os_ansible_distribution_major_ver | int >= 4)
+- name: "Set fact about whether guest OS has inbox driver known issue"
+  ansible.builtin.set_fact:
+    guest_has_driver_known_issue: >-
+      {{
+        esxi_cpu_vendor == 'arm' and
+        (missing_mandatory_drivers | difference(['vmw_pvscsi', 'vmw_balloon']) | length == 0) and
+        (
+         (guest_os_ansible_distribution == 'RedHat' and
+          (guest_os_ansible_distribution_ver is version('10.0', '<=') or
+           guest_os_ansible_distribution_ver is version('9.6', '<='))) or
+         (guest_os_ansible_distribution == 'SLES' and
+          guest_os_ansible_distribution_ver is version('16.0', '<=')) or
+         (guest_os_ansible_distribution == 'Ubuntu' and
+          guest_os_ansible_distribution_ver is version('25.04', '<='))
+        )
+      }}
+
+- name: "Known issue - missing inbox drivers on {{ vm_guest_os_distribution }}"
```

Comment:
All this description `on {{ vm_guest_os_distribution }}` need to be refined and use 'in' is more proper.